SpaceCore/TileSheetExtensions.cs

```diff
@@ -156,22 +156,20 @@ public static void PatchExtendedTileSheet(this IAssetDataForImage asset, Texture
 
         internal static void UpdateReferences()
         {
+            Texture2D oldTS;
             foreach (var asset in TileSheetExtensions.ExtendedTextureAssets)
             {
-                TileSheetExtensions.ExtendedTextures.Remove(asset.Value.BaseTileSheet);
+                oldTS = asset.Value.BaseTileSheet;
                 asset.Value.BaseTileSheet = Game1.content.Load<Texture2D>(asset.Key);
-                if (asset.Value.BaseTileSheet == null)
-                {
+                if (asset.Value.BaseTileSheet == null) {
                     Log.Error("WHAT? null " + asset.Key);
-                }
-                else if (!TileSheetExtensions.ExtendedTextures.ContainsKey(asset.Value.BaseTileSheet))
-                {
-                    TileSheetExtensions.ExtendedTextures.Add(asset.Value.BaseTileSheet, asset.Value);
-                }
-                else
-                {
+                    TileSheetExtensions.ExtendedTextures.Remove(oldTS);
+                    oldTS.Dispose();
+                    }
+                else {
                     TileSheetExtensions.ExtendedTextures[asset.Value.BaseTileSheet] = asset.Value;
-                }
+                    if (oldTS != asset.Value.BaseTileSheet) oldTS.Dispose();
+                    }
```

Comment:
Optimized this because it's being called _every single tick_.

- We do not `.Remove` dict entries unless necessary (in the original code, it was removed then readded back if BaseTileSheet was not null)
- If the BaseTileSheet changes, we invoke `.Dispose` on the old tilesheet to quickly release unmanaged resources
- One unnecessary `if` is removed; the `[...]` syntax already does "adding-or-replacing"